File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.ts
function countKDifference(nums: number[], k: number): number {
    const hashMap: Map<number, number> = new Map<number, number>()
    var ans: number = 0
    
    nums.forEach((value:number) => {
        var add: number = value+k
        var minus: number = value-k
        if (hashMap.has(add)) {
            ans += hashMap.get(add)
        }
        if (hashMap.has(minus)) {
            ans += hashMap.get(minus)
        }
        if (hashMap.has(value)) {
            hashMap.set(value, hashMap.get(value)+1)
        } else {
            hashMap.set(value, 1)
        }
    })
    return ans
};

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.go
func countKDifference(nums []int, k int) int {
    hashMap := make(map[int]int)
    ans := 0
    for i := range nums {
        add := nums[i]+k
        minus := nums[i]-k
        if v, ok := hashMap[add]; ok {
            ans += v
        }
        if v, ok := hashMap[minus]; ok {
            ans += v
        }
        if v, ok := hashMap[nums[i]]; ok {
            v++
            hashMap[nums[i]] = v
        } else {
            hashMap[nums[i]] = 1
        }
    }
    return ans
}